### PR TITLE
Make garbage collection on the next tick to please Node 8.4

### PR DESCRIPTION
**Pull request by kkoopa**
I do not know why this is necessary, nor if it is the correct fix or simply addresses a symptom. My first guess would be something related to TurboFan. Please take a look.

CI:
https://travis-ci.org/nodejs/nan/builds/265943908
https://ci.appveyor.com/project/RodVagg/nan/build/1348

### CODE REVIEW

**Comment by bnoordhuis** [comment]
Does the test pass when you start node with `--single_threaded`?

**Comment by kkoopa** [comment]
No, the existing test does not pass with `--single_threaded` set.

**Comment by kkoopa** [comment]
Seems 8.3 had the same problem, but 8.2.1 is fine.

**Comment by bnoordhuis** [comment]
I had a look but that test does something unsafe: it calls into JS land from a weak callback.

I'd say it's not a matter of figuring out why it doesn't work with v8.4.0 but why it worked at all in older versions. :-)

File: test/js/weak-test.js [hunk 1]
```diff
@@ -21,11 +21,16 @@ test('weak', function (t) {
     count++;
   });
 
-  // run weak callback, should dispose
-  gc();
+  var timeout = setTimeout(function () {
```
Comment (bnoordhuis) [inline test/js/weak-test.js]: I think just wrapping it in a `setImmediate()` should be enough, no need to fuss around with clearing timers again.

Comment (kkoopa) [inline test/js/weak-test.js]: Probably. I copied it from the weak2-test, which tests the internal field variety of weak callbacks. I'll try to change that to `setImmediate` as well.

**Comment by kkoopa** [comment]
Calling into js ought to be fine at that point, no? All callbacks are set as two-pass callbacks and the registered callback runs in the second pass. It should not be doing anything with the handle which was freed either.

**Comment by bnoordhuis** [comment]
I don't think V8 ever made the guarantee that it's safe to call into JS from the GC post-processing step.  I know we used to do it at one time in node.js and we ran into issues after a while.

**Comment by kkoopa** [comment]
Alright. Thanks for looking into this. I am content with having the tests working again.

**Comment by kkoopa** [comment]
Okay, it was `setTimeout` because 0.8 does not have `setImmediate`. Should we just stop testing on 0.8 and release a new minor?

**Comment by bnoordhuis** [comment]
Oh, good point.  I'm way past caring about v0.8 but if we're officially going to drop support for it, it should be semver-major.  If you simply want to stop testing for it, that could be minor or patch.

**Comment by kkoopa** [comment]
We have #676 as a related issue, but it is still seems to be hanging in the air. I think I'll just change this back to `setTimeout` for now and deal with #676 at some later point.

**Comment by bnoordhuis** [comment]
Sounds good.

**Pull request merged**